packages/opentelemetry-core/src/internal/validators.ts

```diff
@@ -14,7 +14,11 @@
  * limitations under the License.
  */
 
-const VALID_KEY_REGEX = /^[a-z][_0-9a-z-*/]{0,255}$/;
+const VALID_KEY_CHAR_RANGE = '[_0-9a-z-*/]';
+const VALID_KEY_REGEX = new RegExp(`^[a-z]${VALID_KEY_CHAR_RANGE}{0,255}$`);
+const VALID_KEY_VENDOR_REGEX = new RegExp(
+  `^[a-z0-9]${VALID_KEY_CHAR_RANGE}{0,240}@[a-z]${VALID_KEY_CHAR_RANGE}{0,13}$`
```

Comment:
vendor keys can start with `[0-9]` but regular keys can't?

Reply:
This is how I interpret the W3C spec:
```
key = lcalpha 0*255( lcalpha / DIGIT / "_" / "-"/ "*" / "/" )
key = ( lcalpha / DIGIT ) 0*240( lcalpha / DIGIT / "_" / "-"/ "*" / "/" ) "@" lcalpha 0*13( lcalpha / DIGIT / "_" / "-"/ "*" / "/" )
lcalpha    = %x61-7A ; a-z
```